pkg/boot/fit/vfit.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+//
+// Performs signature checks on FDT images.
+// Currently supports PGP and raw PKCS1v15 RSA signatures.
+//
+// Expected FDT Format:
```

Comment:
(Quite possibly overkill) Do you have control over the "detached signature" binary format for this scenario? Have you considered something like a pkcs7? That way, the signature can tell you what the hash and signing scheme are (for RSA, either PKCS1.5 or PSS are common; you can also support ECC).

Depending on how this system is intended to grow over time, you may find yourself re-inventing pkcs7. I see a lot of parsing code below to do with gathering up all the metadata needed to validate the signature. (Not to mention, I assume the tools to generate and format the signatures are also nontrivial)

If this format is already designed/published, have you considered building and publishing separate library for parsing and handling those? That would make it easier on the tooling end, as well.

Reply:
I will follow up to see if the RAW signature can be switched to a PGP detached.

Reply:
Unfortunately the format here is already published and needs to be backwards compatible for a tail end of machines.
Understood, I have only noticed internal tools capable of cleanly doing this pkcs signing and exporting a pgp key. That was why I wanted this initial commit to include PGP for future projects.

Reply:
Understood, thanks for the explanation.